src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6693,10 +6693,15 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
         this->visit_expr(*x.m_value);
 
-        // If tmp is a statement and not an expression
-        // never cast into expression using ASRUtils::EXPR
-        // Just ignore and exit the function naturally.
-        if( tmp && !ASR::is_a<ASR::stmt_t>(*tmp) ) {
+        if (eval_count > 0) {
+            // In Interactive mode
+            if ((tmp) && (!ASR::is_a<ASR::expr_t>(*tmp))) {
+                tmp = nullptr;
+            }
```

Comment:
I am a bit confused here. We make `tmp` as `nullptr` everytime it is not an expression. But, why do we do so? I think this means, if `tmp` is a statement, then we are skipping it. Any specific reason to skip statements @Vipul-Cariappa ?

Reply:
If `tmp` is an expression then we return it. So that we can print it. The `pass_wrap_global_stmts` function uses this `tmp` expression and returns the evaluated result of the expression for us to print it later using `PythonCompiler::evaluate` and `interactive_python_repl`.
If `tmp` is a statement, for example, an assignment `x = 5` then we would not want to print anything, so we set `tmp` to `nullptr`.

Reply:
>x = 5 then we would not want to print anything, so we set tmp to nullptr

Although we do not want to print anything, but we do want `x` to be updated to `5`. I am unsure, but if we skip the whole `x = 5` statement, then would the value of `x` be updated to `5`?

Reply:
>The pass_wrap_global_stmts function uses this tmp expression and returns the evaluated result of the expression for us

Does it actually return an evaluated result? Or does it simply convert expressions into assignment statements?

Reply:
Yes. The statement will be executed and `x` will be updated to `5`.

```bash
❯ lp
>>> x: i32
>>> x
0
>>> x = 5
>>> x
5
```

Reply:
> Does it actually return an evaluated result? Or does it simply convert expressions into assignment statements?

It returns the evaluated result. You can run the interactive in verbose mode with `-v` flag to see the AST, ASR, and the LLVM IR generated. Doing so, you can see that it returns the result.

Reply:
With the current approach, I doubt that we are/would-be skipping some stmts like `empty()` and `c_p_pointer()` 

https://github.com/lcompilers/lpython/blob/13bb54eb8243227603add982f3e698eea0143c1d/src/lpython/semantics/python_ast_to_asr.cpp#L5220-L5221

See my comment below https://github.com/lcompilers/lpython/pull/2716/files#r1614616838 which I think would avoid this.